lib/internal/streams/writable.js

```diff
@@ -122,7 +122,13 @@ function WritableState(options, stream, isDuplex) {
   // Crypto is kind of old and crusty.  Historically, its default string
   // encoding is 'binary' so we have to make this configurable.
   // Everything else in the universe uses 'utf8', though.
-  this.defaultEncoding = (options && options.defaultEncoding) || 'utf8';
+  const defaultEncoding = options?.defaultEncoding;
+
+  if (defaultEncoding && !Buffer.isEncoding(defaultEncoding)) {
+    throw new ERR_UNKNOWN_ENCODING(defaultEncoding);
+  }
+
+  this.defaultEncoding = defaultEncoding || 'utf8';
```

Comment:
```suggestion
  if (defaultEncoding == null) {
    this.defaultEncoding = 'utf8';
  } else if(Buffer.isEncoding(defaultEncoding)) {
    this.defaultEncoding = defaultEncoding;
  } else {
    throw new ERR_UNKNOWN_ENCODING(defaultEncoding);
  }
```

Reply:
this wont throw if defaultEncoding is null, so if we want to cover that case I could do something like:


```
const defaultEncoding = options?.defaultEncoding;
  if (defaultEncoding === undefined) {
    this.defaultEncoding = 'utf8';
  } else if (Buffer.isEncoding(defaultEncoding)) {
    this.defaultEncoding = defaultEncoding;
  } else {
    throw new ERR_UNKNOWN_ENCODING(defaultEncoding);
  }
  ```

Reply:
Alternatively we could let `null` be a valid value to get the default? That's what we do in most places in Node.js.

Reply:
changed to your suggestion